src/Provider/CrayfishProvider.php

```diff
@@ -162,36 +171,77 @@ public function boot(Application $app)
-   */
+     * Part of ControllerProviderInterface
+     */
     public function connect(Application $app)
     {
-        $ResourceControllers = $app['controllers_factory'];
+        $controllers = $app['controllers_factory'];
         //
         // Define routing referring to controller services
         //
-        $ResourceControllers
-            ->convert('id', $app['islandora.idToUri'])
-            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
+        $controllers
             ->before($app['islandora.hostHeaderNormalize'])
             ->before($app['islandora.htmlHeaderToTurtle'])
             ->value('id', "");
     
-    
-        $ResourceControllers->get("/resource/{id}/{child}", "islandora.resourcecontroller:get")
+        # ResourceService routes.
+        $controllers->get("/resource/{id}/{child}", "islandora.resourcecontroller:get")
+            ->convert('id', $app['islandora.idToUri'])
+            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
             ->value('child', "")
             ->bind('islandora.resourceGet');
-        $ResourceControllers->post("/resource/{id}", "islandora.resourcecontroller:post")
+        $controllers->post("/resource/{id}", "islandora.resourcecontroller:post")
+            ->convert('id', $app['islandora.idToUri'])
+            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
             ->bind('islandora.resourcePost');
-        $ResourceControllers->put("/resource/{id}/{child}", "islandora.resourcecontroller:put")
+        $controllers->put("/resource/{id}/{child}", "islandora.resourcecontroller:put")
+            ->convert('id', $app['islandora.idToUri'])
+            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
             ->value('child', "")
             ->bind('islandora.resourcePut');
-        $ResourceControllers->patch("/resource/{id}/{child}", "islandora.resourcecontroller:patch")
+        $controllers->patch("/resource/{id}/{child}", "islandora.resourcecontroller:patch")
+            ->convert('id', $app['islandora.idToUri'])
+            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
             ->value('child', "")
             ->bind('islandora.resourcePatch');
-        $ResourceControllers->delete("/resource/{id}/{child}", "islandora.resourcecontroller:delete")
+        $controllers->delete("/resource/{id}/{child}", "islandora.resourcecontroller:delete")
+            ->convert('id', $app['islandora.idToUri'])
+            ->assert('id', $app['config']['islandora']['resourceIdRegex'])
             ->value('child', "")
             ->bind('islandora.resourceDelete');
-        return $ResourceControllers;
+        
+        # TransactionService routes.
+        $controllers->get("/transaction/{id}", "islandora.resourcecontroller:get")
+            ->value('id', "")
+            ->value('child', "")
+            ->before(
+                function (Request $request) {
```

Comment:
This needs to be a local container. Future PR: Inline is complex to maintain and/or debug